### PR TITLE
[ADD]  Contactless payments

### PR DESCRIPTION
**Pull request by siddharth7815**
Task link: 

https://www.druidoo.io/web?db=druidoo-druidoo-master-349765&token=J39sYB1yfc1NMuVYgIBr#id=1080&action=367&model=project.task&view_type=form&menu_id=237

### CODE REVIEW

File: intercoop_addons/account_reconcile_pos_payments/models/account_bank_statement.py [hunk 1]
```diff
@@ -80,91 +80,146 @@ def _reconcile_pos(self):
                     'The journal %s has no CB Contract number.'
                     ) % rec.journal_id.display_name)
 
+            if rec.journal_id.cb_contactless_matching and not rec.journal_id.cb_contract_number_contactless:
+                raise UserError(_(
+                    'The journal %s has no CT Contract number.'
+                ) % rec.journal_id.display_name)
```
Comment (ivantodorovich) [inline intercoop_addons/account_reconcile_pos_payments/models/account_bank_statement.py:86]:
```suggestion
            if rec.journal_id.cb_contactless_matching and not rec.journal_id.cb_contract_number_contactless:
                raise UserError(_(
                    'The journal %s has no CT Contract number.'
                ) % rec.journal_id.display_name)
```

I think it's better to move this validation to a constraint in the `account.journal` model
If you add that constraint you make sure that the contract_number is set when contactless_matching is enabled, and you don't need to check it here

**Review by ivantodorovich — COMMENTED** [review]
Thanks, code looks good! 
I'd like to test it functionally before merging, though

File: intercoop_addons/account_reconcile_pos_payments/models/account_bank_statement.py (hunk 1 above)
Comment (ivantodorovich) [reply intercoop_addons/account_reconcile_pos_payments/models/account_bank_statement.py:86]: Hi @siddharth7815 have you seen this comment?

Comment (siddharth7815) [reply intercoop_addons/account_reconcile_pos_payments/models/account_bank_statement.py:86]: ok, Do you want me to fix it and create new PR for versoin 9.0 or Version 12.0?

Comment (ivantodorovich) [reply intercoop_addons/account_reconcile_pos_payments/models/account_bank_statement.py:86]: Change it in 12.0, this is ok for 9.0

**Pull request merged**